_posts/asyncapi-tool-in-our-toolbox.md

```diff
@@ -0,0 +1,94 @@
+
+A NuGet package alone might not be enough. We could be in a situation where some documentation detailing the reasons for the events or their business meaning might still be something teams need. A documentation repository thus becomes an essential governance part.
+
+Publishers can also decide to follow [Semantic Versioning (SemVer)](https://semver.org) to help packages users understand contracts evolution.
+
+Sharing contracts as NuGet packages containing .NET assemblies comes with a downside, though. It creates platform coupling. Subscribers need to be .NET projects to reference a .NET assembly.
+
+What if one endpoint uses NServiceBus, and another Ruby or Python? [NServiceBus supports cross-platform integration via native message processing](https://particular.net/blog/cross-platform-integration-with-nservicebus-native-message-processing). However, we still need to share contracts in some way. And as said, sharing a .NET assembly between Ruby, or Python, and .NET is not an option.
+
+## Services share schema and contract, not class
+
+What publishers share with subscribers using NuGet packages, for example, are schemas and contracts. By chance, we respect the SOA third tenet using C# classes or interfaces to define the mentioned contracts. However, despite being indeed comfortable, that's not a requirement.
+
+[AsyncAPI](https://www.asyncapi.com) is a protocol-agnostic specification designed to describe asynchronous APIs:
+
+```yaml
+asyncapi: 2.2.0
+info:
+  title: Account Service
+  version: 1.0.0
+  description: This service is in charge of processing user signups
+channels:
+  user/signedup:
+    subscribe:
+      message:
+        $ref:'#/components/messages/UserSignedUp'
+components:
+  messages:
+    UserSignedUp:
+      payload:
+        type: object
+        properties:
+          displayName:
+            type: string
+            description: Name of the user
+          email:
+            type: string
+            format: email
+            description: Email of the user
+```
+
+(from AsyncAPI's samples)
+
+The presented YAML document describes the `UserSignedUp` event much better than what a NuGet package with a C# class and accompanying documentation can do. If not only, for the simple fact that everything is in the same place, the document.
+
+The second interesting aspect of the AsyncAPI specification is the community-driven set of generators. For example, there is a [template to generate C# code from an AsyncAPI YAML document](https://github.com/jonaslagoni/asyncapi-quicktype-template). There are also generators to output HTML documents for a documentation website.
+
+The fact that all information is in a single place, the YAML document, and that there are generators available makes it so that teams can share AsyncAPI specs documents instead of C# compiled artifacts. Once consumers and producers reference the YAML files, they can apply the templates to transform them into C# classes or interfaces.

```

Comment:
```suggestion
The fact that all information is in a single place, the YAML document, and that there are generators available allows teams to share AsyncAPI specs documents instead of C# compiled artifacts. Once consumers and producers reference the YAML files, they can apply the templates to transform them into C# classes or interfaces.

```